docs(linter,minifier): always refer as "ES Modules" instead of "ES6 Modules"

sapphi-red opened the pull request:
Nowadays, we normally call them "ES Modules" instead of "ES6 Modules". I made that consistent through the docs.

camc314 commented:
### Merge activity

* **Nov 7, 10:07 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 7, 10:07 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

sapphi-red committed:
docs(linter,minifier): always refer as "ES Modules" instead of "ES6 Modules" (#15409)

Nowadays, we normally call them "ES Modules" instead of "ES6 Modules". I made that consistent through the docs.
